cookiecutter/main.py

```diff
@@ -203,10 +203,12 @@ def cookiecutter(
 class _patch_import_path_for_repo:  # noqa: N801
     def __init__(self, repo_dir: Path | str) -> None:
         self._repo_dir = f"{repo_dir}" if isinstance(repo_dir, Path) else repo_dir
+        self._path: list[str] | None = None  # Intialize _path to None
 
     def __enter__(self) -> None:
         self._path = copy(sys.path)
         sys.path.append(self._repo_dir)
 
-    def __exit__(self, _type, _value, _traceback):  # type: ignore[no-untyped-def]
-        sys.path = self._path
+    def __exit__(self, _type, _value, _traceback) -> None:  # type: ignore[no-untyped-def]
+        if self._path is not None:  # Check if _path is initialized
+            sys.path = self._path
```

Comment:
This check is not needed because `.__enter__()` and `.__exit__()` are part of the context manager contract. No calling code will ever call `.__exit__()` and find that `._path` is None.

This situation can be synthetically created, as you demonstrated in the test you added, but it's not a realistic issue to address.